Compliance Operator v0.1.57 release notes and enhancements added (#50778)
openshift/openshift-docs: modules/compliance-applying-resource-requests-and-limits.adoc

If several different containers (cgroups) want to run on a contended system, workloads with larger CPU requests are allocated more CPU time than workloads with small requests. The memory request is used during Pod scheduling. On a node that uses cgroups v2, the container runtime might use the memory request as a hint to set `memory.min` and `memory.low` values.

If a container attempts to allocate more memory than this limit, the Linux kernel out-of-memory subsystem activates and intervenes by stopping one of the processes in the container that tried to allocate memory. The memory limit for the Pod or container can also apply to pages in memory-backed volumes, such as an emptyDir.

The kubelet tracks `tmpfs` `emptyDir` volumes as container memory is used, rather than as local ephemeral storage. If a container exceeds its memory request and the node that it runs on becomes short of memory overall, the Pod container might be evicted.

[IMPORTANT]
====

> A container may or may not be allowed to exceed its CPU limit for extended periods. Container runtimes does not stop Pods or containers for excessive CPU usage. To determine whether a container cannot be scheduled or is being killed due to resource limits, see xref:../../security/compliance_operator/compliance-operator-troubleshooting.adoc#compliance-operator-troubleshooting[Troubleshooting the Compliance Operator].

Reviewer: ```suggestion
A container may not exceed its CPU limit for extended periods. Container run times do not stop Pods or containers for excessive CPU usage. To determine whether a container cannot be scheduled or is being killed due to resource limits, see xref:../../security/compliance_operator/compliance-operator-troubleshooting.adoc#compliance-operator-troubleshooting[Troubleshooting the Compliance Operator].
```

// Module included in the following assemblies:
//
// * security/compliance_operator/compliance-scans.adoc

:_content-type: CONCEPT
[id="compliance-applying-resource-requests-and-limits"]

> = Applying resource requests and limits

Reviewer 2: This is all correct and useful information, but I wonder if instead of being part of the CO docs we could have all this info in some generic OCP doc? All that is described here applies equally well outside the CO world..

Author: That is definitely part of the plan. William and I are part of the larger Operator and OSDK docs team as well. The plan, for now, is to get this content available to CO users while we determine where it can also be visible in the Operator/OSDK docs too. Ultimately, it will be a one-line change to a file to insert this module into the Operator/OSDK docs when the time comes!
